hacheck/checker.py

```diff
@@ -66,9 +66,13 @@ def timed_out(*args, **kwargs):
 # Do not cache spool checks
 @tornado.concurrent.return_future
 def check_spool(service_name, port, query, io_loop, callback, query_params, headers):
-    up, extra_info = spool.is_up(service_name)
+    up, extra_info = spool.is_up(service_name, port=port)
     if not up:
         info_string = 'Service %s in down state' % (extra_info['service'],)
+        if extra_info.get('creation', ''):
```

Comment:
Nit, to avoid issues with falsey values:

    if extra_info.get('creation') is not None:
        ...